src/internal/modules/charge-information/lib/helpers.js

```diff
@@ -140,11 +155,81 @@ const isOverridingChargeVersion = async (request, draftChargeVersionStartDate) =
   return !!chargeVersions.find(version => version.dateRange.startDate === draftChargeVersionStartDate);
 };
 
+const processElements = (request, id, selectedElementIds) => {
+  const { chargeVersionWorkflowId } = request.query;
+  const { licenceId, elementId = id } = request.params;
+  const { draftChargeInformation: currentState } = request.pre;
+
+  // Move all newly selected to charge purposes
+  const data = currentState.chargeElements.reduce((acc, element) => {
+    if (element.scheme === 'alcs' && (!selectedElementIds || selectedElementIds.includes(element.id))) {
+      acc.chargePurposes.push({ ...element, scheme: 'sroc' });
+    } else if (element.scheme === 'sroc' && element.id === elementId) {
+      const { chargePurposes, ...chargeElement } = element;
+      acc.chargeElements.push(chargeElement);
+      chargePurposes.forEach(purpose => {
+        if (selectedElementIds && !selectedElementIds.includes(purpose.id)) {
+          acc.chargeElements.push({ ...purpose, scheme: 'alcs' });
+        } else {
+          acc.chargePurposes.push(purpose);
+        }
+      });
+    } else {
+      acc.chargeElements.push(element);
+    }
+    return acc;
+  }, { chargeElements: [], chargePurposes: [] });
+
+  // Move all the newly unselected to charge elements
+  const action = actions.updateChargeCategory(elementId, data.chargeElements, data.chargePurposes);
+  const nextState = reducer(currentState, action);
+  request.setDraftChargeInformation(licenceId, chargeVersionWorkflowId, nextState);
+};
+
+const findChargeCategory = async chargeElement => {
+  const keys = ['source', 'loss', 'isRestrictedSource', 'waterModel', 'volume'];
+  const chargeCategory = await services.water.chargeCategories.getChargeCategory(pick(chargeElement, keys));
+  return {
+    id: chargeCategory.billingChargeCategoryId,
+    reference: chargeCategory.reference,
+    shortDescription: chargeCategory.shortDescription
+  };
+};
+
+const restoreChargeElements = async chargeElements => {
+  // filter out incomplete charge elements when they have used the back button
+  const completeChargeElements = await Promise.all(chargeElements
+    .filter(element => !element.status)
+    .map(async element => {
+      if (element.scheme === 'sroc') {
+        element.chargeCategory = await findChargeCategory(element);
```

Comment:
I don't think you need to lookup the charge category here because the pre-handler on the getCheckAnswers route will do that.